controllers/resource/resource_generator.go

```diff
@@ -369,3 +358,143 @@ func (r *resourceGenerator) CreateDoguSecret(doguResource *k8sv1.Dogu, stringDat
 func wrapControllerReferenceError(err error) error {
 	return fmt.Errorf("failed to set controller reference: %w", err)
 }
+
+type podSpecBuilder struct {
```

Comment:
Maybe move the `podSpecBuilder` to a separate file in this package for a better overview

Reply:
done.